cms/grading/__init__.py

```diff
@@ -347,6 +350,22 @@ def format_status_text(status, translator=None):
                      "text: %r", status, exc_info=True)
         return translator("N/A")
 
+def AIOTranslator(translator):
+    def simplify_status_text(status):
+        mapping_startswith = {
+            "Execution failed because of sandbox error"             : "Judge error, please notify judges",
+            "Execution killed because of forbidden syscall"         : "Program killed due to illegal operation, double check rules regarding allowed system calls",
+            "Execution timed out (wall clock limit exceeded)"       : "Time limit exceeded",
+            "Execution timed out"                                   : "Time limit exceeded",
+            "Execution killed because of forbidden file access:"    : "Forbidden file access, check input/output filenames",
+            "Execution killed with signal"                          : "Program crashed (possibly out of memory)",
```

Comment:
"(possibly out of memory)" feels super misleading here; this is the AIO, they're not going to run out of memory. 99 times out of 100 it'll be a segfault or div-by-zero. If you feel like a little bit of extra work, do you want to map the signal part from SIGFPE/SIGSEGV+SIGBUS/<catchall> to the corresponding "Program crashed after a division by zero error", "Program crashed after accessing or requesting invalid memory" or "Program crashed for an unknown reason"?

If not I'd just nuke the (possibly out of memory) part since it's almost guaranteed to never happen in the AIO.